interfaces/builtin/gpio.go

```diff
@@ -89,6 +91,13 @@ func (iface *gpioInterface) AppArmorConnectedPlug(spec *apparmor.Specification,
 	// requires symlinks to be dereferenced, evaluate the GPIO
 	// path and add the correct absolute path to the AppArmor snippet.
 	dereferencedPath, err := evalSymlinks(path)
+	if err != nil && os.IsNotExist(err) {
+		// If the specific gpio is not available there is no point
+		// exporting it, we should also not fail because this
+		// will block snapd updates (LP: 1866424)
+		logger.Noticef("cannot export not existing gpio %s", path)
+		return nil
+	}
```

Comment:
From a security POV, this looks ok since we are failing closed (the path can't be dereferenced so we don't add it to the snippet).